src/devices/proformtreadmill/proformtreadmill.cpp

```diff
@@ -834,6 +835,99 @@ void proformtreadmill::update() {
             if (counterPoll > 5) {
                 counterPoll = 0;
             }
+        } else if (proform_treadmill_705_cst) {
+            uint8_t noOpData1[] = {0xfe, 0x02, 0x17, 0x03};
+            uint8_t noOpData2[] = {0x00, 0x12, 0x02, 0x04, 0x02, 0x13, 0x04, 0x13, 0x02, 0x00, 0x0d, 0x1b, 0x94, 0x31, 0x00, 0x00, 0x40, 0x50, 0x00, 0x80};
+            uint8_t noOpData3[] = {0xff, 0x05, 0x18, 0x00, 0x00, 0x01, 0x2f, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+            uint8_t noOpData4[] = {0xfe, 0x02, 0x17, 0x03};
+            uint8_t noOpData5[] = {0x00, 0x12, 0x02, 0x04, 0x02, 0x13, 0x04, 0x13, 0x02, 0x00, 0x0d, 0x80, 0x0a, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+            uint8_t noOpData6[] = {0xff, 0x05, 0x00, 0x00, 0x00, 0x84, 0x74, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+            switch (counterPoll) {
+            case 0:
+                writeCharacteristic(noOpData1, sizeof(noOpData1), QStringLiteral("noOp"));
+                break;
+            case 1:
+                writeCharacteristic(noOpData2, sizeof(noOpData2), QStringLiteral("noOp"));
+                break;
+            case 2:
+                writeCharacteristic(noOpData3, sizeof(noOpData3), QStringLiteral("noOp"), false, true);
+                break;
+            case 3:
+                writeCharacteristic(noOpData4, sizeof(noOpData4), QStringLiteral("noOp"));
+                break;
+            case 4:
+                writeCharacteristic(noOpData5, sizeof(noOpData5), QStringLiteral("noOp"));
+                break;
+            case 5:
+                writeCharacteristic(noOpData6, sizeof(noOpData6), QStringLiteral("noOp"));
+
+                if (requestInclination != -100) {
+                    if (requestInclination < 0)
+                        requestInclination = 0;
+                    if (requestInclination != currentInclination().value() && requestInclination >= 0 &&
+                        requestInclination <= 15) {
+                        emit debug(QStringLiteral("writing incline = ") + QString::number(requestInclination));
+                        forceIncline(requestInclination);
+                    }
+                    requestInclination = -100;
+                }
+
+//              Request start does not seem to get initiated automatically.
+//              Instead, trigger it when the treadmill is stopped and a new speed is set via the application.
+//              Note: Setting a speed directly on the treadmill will not work until the initiate commands are sent.
+                if (Distance.value() == 0 && currentSpeed().value() == 0 && requestSpeed > 0) {
```

Comment:
@Novamusix why also adding here distance.value() == 0 ?

it couldn't be useful without distance == 0 so you can resume it after a pause for example?

Reply:
To validate my assumption, I compiled a version that removes distance.value() == 0.

**The following behaviour is observed:**
1. Start a treadmill session, set the inclination to 5, and set the speed to 6 km/h.
2. Pressing the Stop button on the console once will pause the treadmill session.
3. On a connected app, send a new speed request.
4. QZ will detect the current treadmill speed as zero and the requested speed as greater than zero.
5. QZ will trigger send the initiate command, which resets the treadmill timer, sets the inclination to 1, and sets the requested speed.

Previously with the check in place, steps 4 and 5 would be ignored, requiring me to intentionally press the start button to resume the session. With out having a check in place, this could potentially be dangerous if pausing to adjust clothing or stepping half off etc, only for the treadmill to start up again uncontrolled.

Ideally, we would monitor the stop button to determine the treadmill status, but I didn't have any luck with this. It's something I can explore further when I can dedicate some quality time, probably next month.

Reply:
ok thanks, I understand the safety reason. Let's leave like this! Thanks for the tests